torch/csrc/jit/python/pybind_utils.h

```diff
@@ -873,11 +873,37 @@ struct VISIBILITY_HIDDEN tuple_slice {
   int64_t e;
 };
 
+inline IValue fakeScriptObjectToIValue(
+    const c10::FunctionSchema& schema,
+    size_t argumentPosition,
+    py::handle object) {
+  auto argument = schema.arguments().at(argumentPosition);
+  auto class_type = argument.real_type()->expect<c10::ClassType>();
+  auto fake_class_registry =
+      py::module::import("torch._library.fake_class_registry");
+  auto fake_class = fake_class_registry.attr("find_fake_class")(
+      class_type->name().value().qualifiedName());
+  if (!py::isinstance(object.attr("wrapped_obj"), fake_class)) {
+    throw schema_match_error(c10::str(
+        schema.formatTypeMismatchMsg(
+            argument,
+            friendlyTypeName(object),
+            argumentPosition,
+            py::repr(object.attr("wrapped_obj"))),
+        "\nCast error details: ",
+        argument.name(),
+        " is expected to be a FakeScriptObject of ",
+        class_type->name().value().qualifiedName()));
+  }
+  return toIValue(object, PyObjectType::get());
+}
```

Comment:
I don't think we should be implementing createStackForSchema like this -- someone is going to set the allow_fake_script_boject = True and not realize that it should only be used for checking.

Instead we want a separate `bool matchesSchema(schema, args, kwargs)` function that can handle FakeScriptObject